src/lang-promql/promql.ts

```diff
@@ -102,10 +105,12 @@ export class PromQLExtension {
     return this;
   }
 
-  asExtension(): Extension {
-    let extension: Extension = [promQLLanguage];
+  asExtension(fullPromQLLanguage = true): Extension {
```

Comment:
I would avoid a boolean option here, also because we may want to add more sub-types (label names, label values). Maybe a `LanguageType` enum with values `PromQL` and `MetricName` instead?

Reply:
mmm perhaps later. It would require more thought to integrate it properly and for the moment it doesn't worth it I think (I mean to manage the LanguageType in the complete / linter)